OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -194,6 +198,21 @@ public override bool Tick(Actor self)
 					if ((loc - searchFromLoc.Value).LengthSquared > searchRadiusSquared)
 						return int.MaxValue;
 
+					// Add a cost modifier to harvestable cells to prefer resources that are closer to the refinery.
+					// This reduces the tendancy for harvesters to move in straight lines
+					if (procPos.HasValue && harvInfo.ResourceRefineryDirectionCostModifier > 0 && harv.CanHarvestCell(self, loc))
```

Comment:
Why only have the directional cost for cells that contain resources?

Reply:
The goal here is to add the cost only once, to bias the choice between two cells that are otherwise the same cost. Applying this to all cells in the path would introduce a distance dependence, and make it hard to reason about what the harvester behaviour will be.

Reply:
The problem is that this might bias the harvester to cross over to a non-continuous patch when that is closer to the refinery, even when a harvestable cell is right next to it. I guess that is only a problem if you allow negative cost modifiers though.

Reply:
The modifier adds +/- 1 cell equivalent cost with the default setup. I don't see how this could cause it to transition fields?

Reply:
Consider this:
1. Resource cell next to the harvester but opposite the refinery. Cost in cells: 1 + 1 = 2
2. Resource cell two cells away from the harvester but in the direction of the refinery. Cost in cells: 2 -1 = 1.

When the modifier gets bigger, the gaps get larger (and if I'm not mistaken the default there is quite a bit bigger than one cell).

Reply:
The base pathfinder cost for moving a cell is 10000 / modifier, which evaluates to 125 for harvesters on clear terrain, or 177 for diagonal moves. The default `ResourceRefineryDirectionCostModifier` is therefore 0.8 cells (~.56 for diagonals), which is just a bit more than needed to give a 1 cell cost difference for opposing diagonal tiles. that are diagonally aligned with the refinery (which is the worst contrived case - in practice the cost differences will be less).

The cost modifier is added to all resource cells, so positive vs negative doesn't really matter - only the relative difference. The possibility for having an overall negative path cost is indeed a problem, which I'll fix.

Reply:
Once again, I'm incapable of reading: I kept reading 1000/modifier in the code, which would make the default equivalent to 8 cells instead of 0.8.